src/main/java/seedu/module/logic/commands/AddDeadlineCommand.java

```diff
@@ -49,9 +49,10 @@ public CommandResult execute(Model model) throws CommandException {
 
         TrackedModule moduleToEdit = lastShownList.get(index.getZeroBased());
 
-        ArchivedModule archivedModule = new ArchivedModule(moduleToEdit.getModuleCode(),
-                moduleToEdit.getTitle(), moduleToEdit.getDescription());
-        TrackedModule editedModule = new TrackedModule(archivedModule , deadline);
+        TrackedModule editedModule = new TrackedModule(
+                new ArchivedModule(moduleToEdit.getModuleCode(),
+                moduleToEdit.getTitle(), moduleToEdit.getDescription()),
```

Comment:
This is still a new `ArchivedModule` being created. Can I suggest that we just re-reference the same archived module object. We can expose a `getArchivedModule()` method for `TrackedModule` for example

Reply:
I have modified it to create a setDeadline() method instead of calling ArchievedModule object.